src/main/java/emissary/output/DropOffUtil.java

```diff
@@ -969,25 +974,32 @@ public Date extractEventDateFrom(final IBaseDataObject d, final boolean lastReso
     }
 
     /**
-     * Extracts from the provided {@link IBaseDataObject} the last file extension from each value in the "Original-Filename"
-     * parameter, if that value contains "." before its last character. If one or more file extensions are extracted, the
-     * IBaseDataObject's "FILEXT" parameter is set as the unique set of extracted file extensions, converted to lowercase.
+     * Extracts from the provided {@link IBaseDataObject} the last file extension from each filename, if that value contains
+     * "." before its last few characters. If one or more file extensions are extracted, the IBaseDataObject's "FILEXT"
+     * parameter is set as the unique set of extracted file extensions, converted to lowercase.
      *
      * @param p IBaseDataObject to process
      *
      */
     void extractUniqueFileExtensions(IBaseDataObject p) {
-        if (p.hasParameter(ORIGINAL_FILENAME)) {
-            final Set<String> extensions = new HashSet<>();
-            for (Object filename : p.getParameter(ORIGINAL_FILENAME)) {
-                final String fn = (String) filename;
-                if (StringUtils.isNotEmpty(fn) && fn.lastIndexOf('.') > -1) {
-                    final int pos = fn.lastIndexOf('.') + 1;
-                    if (pos < fn.length()) {
-                        final String fext = fn.substring(pos).toLowerCase();
-                        if (fext.length() > 0 && fext.length() <= this.maxFilextLen) {
-                            extensions.add(fext);
-                        }
+        List<String> filenames = getBestFilenames(p);
+        final Set<String> extensions = new HashSet<>();
+        for (String filename : filenames) {
+
+            // if what we have is a full filepath, extract just the filename (text after the file path separator)
+            Matcher matcher = JUST_FILENAME_PATTERN.matcher(filename);
+            String justFilename = filename;
+            if (matcher.find()) {
+                justFilename = matcher.group(0);
+            }
+
+            // get the text after the last period
+            if (justFilename.lastIndexOf('.') > -1) {
```

Comment:
Consider using [FileUtils.getExtension(...)](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FilenameUtils.html#getExtension(java.lang.String)) and then applying the range/length filtering.  
```suggestion
           final String fext = FileUtils.getExtension(filename);
           if (StringUtils.isNotBlank(fext) && fext.length() <= this.maxFilextLen) {
               extensions.add(fext);
           } 
```